Source/Core/VideoBackends/Vulkan/VKScheduler.h

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <cstddef>
+#include <queue>
+
+#include "CommandBufferManager.h"
+#include "Common/Align.h"
+
+namespace Vulkan
+{
+class Scheduler
```

Comment:
This whole file could use a bunch of documentation. It implements some non-trivial data structures with some unusual memory management choices, and while I think I generally agree with the design, it took me 30min of reading this file to feel like I understood what it was trying to do and why.

Basically, CommandChunk is a sequence of tagged values, but the "tag" happens to be the vtable of each TypedCommand<T>, which is pretty clever. I like it a lot!